File: 알고리즘/최이삭/WorkoutClothes.js
function solution(n, lost, reserve) {
  let lost_obj = {}
  let reserve_obj = {}
  lost.forEach(el => lost_obj[el] = true)
  reserve.forEach(el => reserve_obj[el] = true) 
  lost.forEach(el => {
      if(reserve_obj[el] === true) {
          delete lost_obj[el];
          delete reserve_obj[el];
      }
      else if(reserve_obj[el - 1] === true && lost_obj[el - 1] !== true) {
          delete lost_obj[el];
          delete reserve_obj[el - 1];         
      }
      else if(reserve_obj[el + 1] === true && lost_obj[el + 1] !== true) {
          delete lost_obj[el];
          delete reserve_obj[el + 1];         
      }
  })    
  return n - Object.keys(lost_obj).length;
}

File: 알고리즘/전승민/workoutClothes/workoutClothes.kt
fun workoutClothes(n: Int, lost: IntArray, reserve: IntArray): Int {
    val clothesArr = IntArray(n) {1}
    lost.forEach {
        clothesArr[it-1]--
    }
    reserve.forEach {
        clothesArr[it-1]++
    }
    clothesArr.forEachIndexed { index, i ->
        if(i==0) {
            if(index!=0 && clothesArr[index-1]==2) {
                clothesArr[index-1]--
                clothesArr[index]++
            } else if(index!=clothesArr.size-1 && clothesArr[index+1]==2) {
                clothesArr[index+1]--
                clothesArr[index]++
            }
        }
    }
    return clothesArr.filter { it > 0 }.size
} // https://programmers.co.kr/learn/courses/30/lessons/42862